packages/@react-aria/interactions/src/useFocusVisible.ts

```diff
@@ -135,23 +135,23 @@ function setupGlobalFocusEvents(element?: HTMLElement | null) {
     focus.apply(this, arguments as unknown as [options?: FocusOptions | undefined]);
   };
 
-  documentObject.addEventListener('keydown', handleKeyboardEvent, true);
-  documentObject.addEventListener('keyup', handleKeyboardEvent, true);
-  documentObject.addEventListener('click', handleClickEvent, true);
+  documentObject?.addEventListener('keydown', handleKeyboardEvent as EventListener, true);
```

Comment:
why did the cast become necessary?

Reply:
It is necessary because there is a gap between specific event types (e.g., `KeyboardEvent`) and the `EventListener` type expected by `addEventListener`. I updated the approach to use [createEventListener](https://github.com/adobe/react-spectrum/pull/6046/commits/40cd5dd2e6f88d02680fb847a2500d4bbecff51a#diff-c85095437b8c2976c74228b10139d930515efcd0be5bef41bda3160ba91ac2dbR38-R41) that maintains type safety and existing function signatures without casting.

Reply:
I believe that the approach I mentioned above will cause problems with cleanup, so I updated it [here](https://github.com/adobe/react-spectrum/pull/6046/commits/c5ddb2423ac07abbcd036391218fc23bb65acfee).